frontend-maven-plugin/src/main/java/com/github/eirslett/maven/plugins/frontend/mojo/InstallNodeAndPnpmMojo.java

```diff
@@ -107,6 +104,46 @@ public void execute(FrontendPluginFactory factory) throws Exception {
 
         String validNodeVersion = getDownloadableVersion(nodeVersion);
 
+        // Use different names to avoid confusion with fields `nodeDownloadRoot` and
+        // `pnpmDownloadRoot`.
+        //
+        // TODO: Remove the `downloadRoot` config (with breaking change) to simplify download root
+        // resolution.
+        String resolvedNodeDownloadRoot = getNodeDownloadRoot();
+        String resolvedPnpmDownloadRoot = getPnpmDownloadRoot();
+
+        if (isAtlassianProject(project) &&
+                isBlank(serverId) &&
+                (isBlank(resolvedNodeDownloadRoot) || isBlank(resolvedPnpmDownloadRoot))
+        ) { // If they're overridden the settings, they be the boss
+            getLog().info("Atlassian project detected, going to use the internal mirrors (requires VPN)");
+
+            serverId = "maven-atlassian-com";
+            try {
+                install(factory, validNodeVersion,
```

Comment:
similarly here - maybe we can just do a single install at the bottom with no fallbacks? simplifies this a lot too

Reply:
I'll resolve this and we'll reply in the other thread since it's the same for everything